vignettes/simplify_function.Rmd

```diff
@@ -1,28 +1,121 @@
 ---
-title: "simplify_function"
+title: "Simplify Function"
 output: html_document
 ---
 
 ```{r setup, include=FALSE}
 knitr::opts_chunk$set(echo = TRUE)
 ```
 
-## R Markdown
+## Overview
 
-This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.
+Estimator's `predict()` and `evaluate()` provide an easy and efficient way of making predictions and evaluations on new data. However, the prediction and evaluation results are pretty verbose and usually in forms of nested lists, which isn't user-friendly. By default, the results produced from canned estimators will be simplified using a default simplify function for both `predict()` and `evaluate()`. You can also write a custom simplify function to them. Custom estimators do not simplify results by default unless you provide a custom simplify function. In this tutorial, we'll illustrate how the simplified results look like and how you can write your own simplify function.
 
-When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:
+## Simplify Prediction Results
 
-```{r cars}
-summary(cars)
+First of all, let's define our feature columns, input function, and a basic linear classifier:
+
+```{r}
+library(tfestimators)
+
+cols <- feature_columns( 
+  column_numeric("disp", "cyl")
+)
+
+mtcars_input_fn <- function(data) {
+  input_fn(data, 
+           features = c("disp", "cyl"), 
+           response = "vs")
+}
+
+indices <- sample(1:nrow(mtcars), size = 0.80 * nrow(mtcars))
+train_data <- mtcars[indices, ]
+test_data  <- mtcars[-indices, ]
+
+model <- linear_classifier(feature_columns = cols)
+```
+
+and we train the model using the input function constructed from training data:
+
+```{r}
+model %>% train(mtcars_input_fn(train_data))
+```
+
+Here's how the prediction output looks like without applying the simplifier on the results:
+
+```{r}
+model %>% predict(mtcars_input_fn(mtcars[1, ]), simplify = FALSE)
+```
+
+Note that here we only evaluate the first row of `mtcars` data since the results will be in forms of long nested lists that may not fit in this page.
+
+
+### Default Simplifier
+
+If `simplify = TRUE`, a default simplify function for predictions will be used to flatten the results so they are more appearing and concise like the following:
```

Comment:
We should paste the output directly into the vignette so that the vignettes can be built on a system without TensorFlow (and also so that they build faster)